network/f5/bigip_sysdb.py

```diff
@@ -0,0 +1,335 @@
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: bigip_sysdb
+short_description: Manage BIG-IP system database variables
+description:
+   - Manage BIG-IP system database variables
+version_added: "2.0"
+options:
+  connection:
+    description:
+      - The connection used to interface with the BIG-IP
+    required: false
+    default: rest
+    choices: [ "rest", "icontrol" ]
+  server:
+    description:
+      - BIG-IP host
+    required: true
+  key:
+    description:
+      - The database variable to manipulate
+    required: true
+  password:
+    description:
+      - BIG-IP password
+    required: true
+  state:
+    description:
+      - The state of the variable on the system. When C(present), guarantees
+        that an existing variable is set to C(value). When C(reset) sets the
+        variable back to the default value. At least one of value and state
+        C(reset) are required.
+    required: false
+    default: present
+    choices: [ "present", "reset" ]
+  user:
+    description:
+      - BIG-IP username
+    required: true
+  value:
+    description:
+      - The value to set the key to. At least one of value and state C(reset) are required.
+    required: false
+  validate_certs:
+    description:
+      - If C(no), SSL certificates will not be validated. This should only be
+        used on personally controlled sites using self-signed certificates.
+    required: false
+    default: true
+
+notes:
+   - Requires the bigsuds Python package on the host if using the iControl
+     interface. This is as easy as pip install bigsuds
+
+requirements: [ "bigsuds", "requests" ]
+author: Tim Rupp <caphrim007@gmail.com> (@caphrim007)
+'''
+
+EXAMPLES = """
+- name: Set the boot.quiet DB variable on the BIG-IP
+  bigip_sysdb:
+      user: "admin"
+      password: "my_password"
+      server: "big-ip"
+      key: "boot.quiet"
+      value: "disable"
+  delegate_to: localhost
+
+- name: Disable the initial setup screen
+  bigip_sysdb:
+      user: "admin"
+      password: "my_password"
+      server: "big-ip"
+      key: "setup.run"
+      value: "false"
+  delegate_to: localhost
+
+- name: Reset the initial setup screen
+  bigip_sysdb:
+      user: "admin"
+      password: "my_password"
+      server: "big-ip"
+      key: "setup.run"
+      state: "reset"
+  delegate_to: localhost
+"""
+
```

Comment:
hmm, had not considered that the module does not return anything except the common defaults .... not sure how to indicate that (changed should not be documented).